lib/MusicBrainz/Server/Data/Utils.pm

```diff
@@ -612,6 +613,17 @@ sub boolean_from_json {
     return $bool ? 1 : 0;
 }
 
+sub datetime_to_iso8601 {
+    my $date = shift;
+
+    if ($date) {
```

Comment:
I would add this at the top instead:
```Perl
return unless defined $date;
```
Assuming it makes sense to allow passing undef into here, which maybe it does. But if someone were to pass 0 or '', we should probably let it throw an exception and not return the incorrect data back.

Reply:
This seems to mess up with the output somehow.

Reply:
That would be strange...did you add it below the `my $date = shift;`?

Reply:
Yup. It somehow messes up the `TO_JSON` sub and causes keys to become values. and messes with the output.

Reply:
Ah, that's because when you call a function inside a hash definition, like `{ foo => datetime_to_iso8601($self->foo), bar => 1 }`, it calls `datetime_to_iso8601` in *list* context as opposed to *scalar* context. The bare `return` inside `datetime_to_iso8601` returns an empty list in list context, resulting in `{ foo => (), bar => 1 }` which is the same as `{ 'foo', 'bar', 1 }` - that's why the keys become values. (Tip: A `=>` is the same as a comma, except it automatically quotes the key for you.)

There are two solutions:

1. Change the return to `return undef unless defined $date;` (note the added `undef`), so that it returns `(undef,)` in list context instead.
2. Alternatively, call the function in scalar context outside of the hash definition. This can be done by assigning the result of the function to a scalar variable.

Reply:
Wouldn't it be equivalent to `return $date` in case $date in undefined?

Reply:
Yes, but `undef` is not the only falsy value that `$date` can be. Anyway it doesn't matter that much.